lib/User/Backend.php

```diff
@@ -37,9 +37,10 @@
 use OCP\User\Backend\ICustomLogout;
 use OCP\User\Backend\IGetDisplayNameBackend;
 use OCP\User\Backend\IPasswordConfirmationBackend;
+use OCP\User\Backend\ISetDisplayNameBackend;
 use Psr\Log\LoggerInterface;
 
-class Backend extends ABackend implements IPasswordConfirmationBackend, IGetDisplayNameBackend, IApacheBackend, ICustomLogout, ICountUsersBackend {
+class Backend extends ABackend implements IPasswordConfirmationBackend, IGetDisplayNameBackend, ISetDisplayNameBackend, IApacheBackend, ICustomLogout, ICountUsersBackend {
```

Comment:
This would mean that users can change the displayname also trough the Nextcloud profile page, no?

Since the IdP has authority over the display name that is used, we should not implement this interface I think.

Reply:
Can confirm that displayName can be changed trough the Nextcloud profile page when this is implemented (manually patch v30.0.10 with v7.2.0).